demo/src/main/java/com/eis/geoCalendar/demo/ContactsActivity.java

```diff
@@ -1 +1,212 @@
+package com.eis.geoCalendar.demo;
+
+import androidx.fragment.app.DialogFragment;
+import androidx.fragment.app.FragmentActivity;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.ContactsContract;
+import android.view.View;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class handles a friend's list, picking up contacts from the device's address book and saving them into app's sharedPreferences file.
+ * Contacts can be deleted from this list by clicking them.
+ * <p>
+ * Contacts will be displayed as a String in a TextView, inside a LinearLayout of a ScrollView.
+ * The string's format will be #contactName + " " + #contactNumber.
+ * <p>
+ * N.B.: the contact's number is parsed before displaying, to remove possible spaces or parenthesis.
+ *
+ * @author Tonin Alessandra
+ */
+
+public class ContactsActivity extends FragmentActivity implements RemoveFriendDialogFragment.RemoveFriendDialogListener {
```

Comment:
I thought about this, but I had some troubles: the method `getPreferences()` must be accessed from an activity because this retrieves a default shared preference file that belongs to the activity. So still I don't know exactly how to do this, because most of the methods that handle the `sharedPreferences` access, need to use `getPreferences()`.

Reply:
Maybe I should try using `getSharedPreferences() `method instead, which can be called from any Context in the app.

Reply:
Solved this, I moved all `sharedPreferences`' related code to another class.

Reply:
Nice one